### DISCUSSION TITLE
HttpOnly cookies under docker-compose & Cypress

### DISCUSSION THREAD

**Discussion by navalex** [cypress-io/cypress#30106] [CI setup]
Hey guys,

I'm encountering a strange behaviour when trying to use Cypress with the docker image `cypress/included:13.5.0` when my server try to set httpOnly cookies.

For the moment I have only 1 test that try to login a user and logout. Works perfectly on localhost running the test manually with Cypress GUI.

I have several services running at same time to make my app works (front, back, supertokens/auth, 2 postgres, monitoring...), so I tried to make a special CI-dedicated docker-compose file to run my E2E tests. But the test stop pass in this environment.

The only difference I have with my localhost test, is urls. On my CI docker compose, I prefer avoid opening any port, and I just make services communicate using their service name as url (ex: http://backend:3005).

Anyway to gather some help, I created a really basic repo with a frontend and backend server, and a docker-compose to run the whole thing in headless in the same environment as my personal project. The backend set 3 cookies in httpOnly, sameSite none and secure true.

Reproduce issue: https://github.com/navalex/cy-httpOnly-cookie-error

Any idea of why it does that and how to fix it ? I don't want to change my auth system only because Cypress doesn't support that...

**Comment by navalex** [comment]
Any clue of what is happening ?

**Comment by albert9052** [comment]
@navalex  I had a similar issue... Have you found a way to solve this?
